tests/performance_tests/performance_test_basic.py

```diff
@@ -132,27 +149,18 @@ def waitForEmptyBlocks(node):
     guide = log_reader.calcChainGuide(data, numAddlBlocksToPrune)
     tpsStats = log_reader.scoreTransfersPerSecond(data, guide)
     print(f"Blocks Guide: {guide}\nTPS: {tpsStats}")
-    report = log_reader.createJSONReport(guide, tpsStats, args)
+    report = log_reader.createJSONReport(guide, tpsStats, args, completedRun)
     print(report)
     if args.save_json:
         log_reader.exportAsJSON(report, args)
 
-    assert transactionsSent == data.totalTransactions , f"Error: Transactions received: {data.totalTransactions} did not match expected total: {transactionsSent}"
+    if completedRun:
+        assert transactionsSent == data.totalTransactions , f"Error: Transactions received: {data.totalTransactions} did not match expected total: {transactionsSent}"
+    else:
+        os.system("pkill trx_generator")
```

Comment:
This is a candidate for replacing with explicit subprocess control, but we'll wait on that until the regression test harness has support for it.